### PR TITLE
Switch to using Eigen for Quaternion and Matrix.

### LINKED ISSUE #6: collada to urdf does not need to depend on TF
From ros/robot_model#119

This part of the code:
https://github.com/ros/robot_model/blob/19ac39ac7324237485a27664183442a93319a269/collada_urdf/src/collada_to_urdf.cpp#L345

seems to be the only reason collada urdf depends on TF. Removing it or using urdf::Pose instead will remove the dependency.

### PR DESCRIPTION
**Pull request by clalancette**
This should fix #6, though to be conservative we may want to merge this onto a melodic-devel branch, rather than kinetic-devel.

Signed-off-by: Chris Lalancette <clalancette@osrfoundation.org>

### CODE REVIEW

File: collada_urdf/CMakeLists.txt [hunk 1]
```diff
@@ -48,13 +48,33 @@ endif()
 find_package(Boost REQUIRED COMPONENTS system filesystem program_options)
 include_directories(${Boost_INCLUDE_DIR})
 
+# We don't build anything here, but in order to export the right
+# build flags in catkin_package(), we still have to find Eigen3 here.  Note
+# that this is somewhat complicated because of our need to support Ubuntu
+# all the way back to saucy.  First we look for the Eigen3 cmake module
```
Comment (sloretz) [inline collada_urdf/CMakeLists.txt]: [REP 3](http://www.ros.org/reps/rep-0003.html#kinetic-kame-may-2016-may-2021) says kinetic targets xenial and above. [libeigen3-dev](https://packages.ubuntu.com/search?keywords=libeigen3-dev) is in trusty and above. I think this block can be replaced with `find_package(Eigen3 REQUIRED)`

File: collada_urdf/CMakeLists.txt [hunk 2]
```diff
@@ -48,13 +48,33 @@ endif()
...
+# fall-back to the version provided by cmake_modules, which is a stand-in.
+find_package(Eigen3 QUIET)
+if(NOT EIGEN3_FOUND)
+  # saucy does not have Eigen3, but have Eigen instead
+  find_package(cmake_modules REQUIRED)
+  find_package(Eigen REQUIRED)
+  set(EIGEN3_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS})
+endif()
+
+# Note that eigen 3.2 (on Ubuntu Wily) only provides EIGEN3_INCLUDE_DIR,
```
Comment (sloretz) [inline collada_urdf/CMakeLists.txt]: Same here, xenial through bionic is `3.3.*`

**Review by sloretz — COMMENTED** [review]
Sorry it took me so long to review this. Looks good after a couple CMake changes.

**Comment by sloretz** [comment]
I notice #6 has a melodic label. I can't think of anything that would block targeting this at kinetic. Can you think of anything?

**Comment by clalancette** [comment]
> I notice #6 has a melodic label. I can't think of anything that would block targeting this at kinetic. Can you think of anything?

I agree with you; this is all internal code, so we should be able to merge this for kinetic as well.  There is a slight chance for downstream breakage if downstream packages were depending on this one to pull in TF, but I think that is acceptable (and should be relatively easy to fix).

File: collada_urdf/CMakeLists.txt (hunk 1 above)
Comment (clalancette) [reply collada_urdf/CMakeLists.txt]: Ah, good point.  It looks like Kinetic used to support Wily, but we've long stopped worrying about that.  Fixed in 4edadbf

**Review by sloretz — APPROVED** [review]

**Comment by clalancette** [comment]
Great, thanks!  Merging.

**Pull request merged**